src/Microsoft.Management.UI.Internal/Microsoft.PowerShell.GraphicalHost.csproj

```diff
@@ -12,6 +12,66 @@
     <DefineConstants>$(DefineConstants);CORECLR</DefineConstants>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <StronglyTypedOutputFolder>gen</StronglyTypedOutputFolder>
+    <SkipWPFcsproj>$([System.String]::Copy('$(MSBuildProjectFile)').Contains('wpftmp'))</SkipWPFcsproj>
+  </PropertyGroup>
+
+  <Target Name = "CreateDirectories" BeforeTargets = "PSResources">
+    <MakeDir
+       Directories = "$(StronglyTypedOutputFolder)"/>
+  </Target>
+
+  <ItemGroup>
+    <!-- We have to use the workaround for the WPF project.
+        .Net 5 Preview.2 has some issues with WPF projects:
+          - Strong typed resources does not work as expected out of box.
+            (Really they are not generated at all.)
+            So we assign items in 'PSEmbeddedResource' to expected format
+            and explicitly call 'GenerateResource' task from custom 'PSResources' target.
+          - .Net 5 Preview.2 generates a temporory csproj file
+            that duplicates call our custom 'PSResources' target.
+            To avoid this we skip a project with 'wpftmp' in name.
```

Comment:
@iSazonov Thanks for this work! I have 3 concerns:
1. The strongly typed .cs files are not generated during the build, and thus reference to the resource strings like `ExtendedTypeSystem.InvalidCastExceptionWithInnerException` are all shown as errors in VS Code. It's likely the same in Visual Studio, but I didn't try.
2. Is this workaround still needed? We have moved to .NET 5 preview.4.
3. Question: is this really more maintainable than using `ResGen` as before? Adding targets to `.csproj` always scares me. What are the benefits moving to `ResXFileCodeGenerator`?

Reply:
> 1.

Resx files are converted to cs files well for me when I run Start-PSBuild. Or I don't understand your question?

> 2. 

I see no activity in MSBuild and SDK.  They use tricky way to compile WPF but I don't think they are going to change this now.

> 3.

It seems strong typed resources are rarely used in while and in this sense it is less maintainable.

> Adding targets to .csproj always scares me.

I guess you are thinking about daily enhancements. In fact, our compile workflow is stable for years. So why not solve the following:
- Localization #666
I expect we get an increase in the number of users in 10x or 100x in next dev cycle. And user will ask about localization. This change is one step to MSFT team to make this process public and save their resources.
- #5777 
The change is one step to removing dependency on PowerShell to compile the project that helps third-party distributers.
We have still to run TypeGen first. But it is a stable too and we could move the step to csproj too. As result (1) third-party distributers will be able to compile PowerShell without having to install PowerShell, (2) we will be able to compile in VS Code and VS without Start-PSBuild.

Reply:
> Resx files are converted to cs files well for me when I run Start-PSBuild. Or I don't understand your question?

I checked out the `resgen-msbuild-3` branch, did a clean build using `Start-PSBuild -Clean` with dotnet SDK `5.0.100-preview.3.20216.6`.
After that, I couldn't find the `.cs` files generated from the resource files. The only `gen` folder I could find is `src\Microsoft.Management.UI.Internal\gen`.

The following screenshot is what I get after opening `src\System.Management.Automation` folder in VS Code. A lot of errors because the property references to resources don't work anymore.

![image](https://user-images.githubusercontent.com/127450/81721342-96f3ed00-9434-11ea-97f8-6d7d1809f1f5.png)

> It seems strong typed resources are rarely used in while and in this sense it is less maintainable.

Just so I'm clear, you agree that the `ResGen` we use before this PR is more maintainable, right?

> I guess you are thinking about daily enhancements.

I'm not thinking about enhancement, but purely maintenance, e.g. something changed (either dotnet sdk or the PS project itself) and we have to change this target to make it continue to work ...

> So why not solve the following:

It's not clear to me how this will help solving those 2 issues,
- #666 -- we will need some people to translate the resource strings. Whether the resource strings are consumed by `ResGen` or `ResXFileCodeGenerator` doesn't seem to matter.
- #5777 -- this complaint is about building PowerShell using PowerShell. `ResGen` doesn't use PowerShell.

Reply:
> After that, I couldn't find the .cs files generated from the resource files. The only gen folder I could find is src\Microsoft.Management.UI.Internal\gen.

The files are generated in obj\ folder where *.resource are. If Start-PSBuild success you have the files.
VS Code works well for me too (after restart?).

> Just so I'm clear, you agree that the ResGen we use before this PR is more maintainable, right?
> I'm not thinking about enhancement, but purely maintenance

ResGen tool is not changed for years. We don't any maintenance.
.Net team is porting MSBuild step-by-step and it is not high priority for them. But if they did I think they will avoid breaking changes.

> It's not clear to me how this will help solving those 2 issues,

Currently we have to use Build.psm1 module to build PowerShell. Third-party distribution maintainers have to port the module to bash shell to build PowerShell without PowerShell. If we change the module we break their process, they have to port again. If all build logic is in csproj files they get updates automatically.
With the change they should have a workaround only for TypeGen - resource generation now works with `dotnet build`.  Additional benefits - now if we change any resx file it will be automatically compiled without needs to run `Start-PSBuild -ResGen`. We could make the same for TypeGen too.

Reply:
> The files are generated in obj\ folder where *.resource are. If Start-PSBuild success you have the files.
VS Code works well for me too (after restart?).

Found the `.cs` files under `obj\Debug\netcoreapp5.0`, but opening in VS Code still doesn't work -- a lot errors due to the resource string references. @TravisEz13 do you see the same issue?

> Currently we have to use Build.psm1 module to build PowerShell.
Additional benefits - now if we change any resx file it will be automatically compiled without needs to run `Start-PSBuild -ResGen`.

`ResGen` doesn't depend on PowerShell. Anyone wants to build without PowerShell can do it as described in our doc: https://github.com/PowerShell/PowerShell/blob/master/docs/building/internals.md#resgen 

I agree that's a benefit, but personally, I don't think it's convincing ... I personally don't want to incorporate everything into msbuild, as I personally believe a script is more maintainable than black magic in msbuild 😄

Reply:
> black magic in msbuild

I agree (I spent my time on this 😄  ) but we have already second (third?) request to simplify building for third-party distributions. I believe we should help them. I believe building is stable area. In contrast packaging is changing often but this area where they must act independently.
I think MSFT team is interested in broader PowerShell support.


>  but opening in VS Code still doesn't work 

On your screenshot on bottom I see reference to SMA. On my system the refrence is to PowerShell.sln
![image](https://user-images.githubusercontent.com/22290914/81770147-db48c100-94f8-11ea-933a-e965b5f480e7.png)

Reply:
It's the same for me even if I start VS Code at the root of the repo (referencing PowerShell.sln). I'm using the latest VS Code (1.45.0) and latest C# extension.
@TravisEz13 @vexx32 Can you please checkout this branch and give it a try? 

![image](https://user-images.githubusercontent.com/127450/81844646-3e8c2080-9504-11ea-9e67-c872c22eae46.png)

Reply:
I can share
```
dotnet --list-sdks
3.1.201 [C:\Program Files\dotnet\sdk]
3.1.300-preview-015135 [C:\Program Files\dotnet\sdk]
5.0.100-preview.3.20216.6 [C:\Program Files\dotnet\sdk]
5.0.100-preview.3.20216.6 [C:\Users\1\AppData\Local\Microsoft\dotnet\sdk]
5.0.100-preview.4.20229.10 [C:\Users\1\AppData\Local\Microsoft\dotnet\sdk]


 $env:path -split ";"
C:\WINDOWS\system32
C:\WINDOWS
C:\WINDOWS\System32\Wbem
C:\WINDOWS\System32\WindowsPowerShell\v1.0\
C:\Program Files\Git\cmd
C:\Program Files\Microsoft VS Code\bin
C:\WINDOWS\System32\OpenSSH\
C:\Users\1\AppData\Local\Microsoft\dotnet\
C:\Program Files\dotnet\
C:\Users\1\AppData\Local\Microsoft\dotnet
C:\Users\1\AppData\Local\Microsoft\powershell
C:\Users\1\AppData\Local\Microsoft\WindowsApps
C:\Program Files\Microsoft VS Code\bin
C:\Users\1\AppData\Local\Microsoft\WindowsApps
C:\Users\1\.dotnet\tools
C:\Users\1\AppData\Local\GitHubDesktop\bin
```

Reply:
Also Visual Studio 2019 16.6 Preview6 is installed.

Reply:
```
OmniSharp server started.
    Path: c:\Users\1\.vscode\extensions\ms-dotnettools.csharp-1.21.18\.omnisharp\1.35.1\OmniSharp.exe
```

Reply:
I'm seeing the same errors as @daxian-dbw. I wonder if Visual Studio is an assumed dependency for these new libs for some reason?

Reply:
Mystics :-)
I rebased and again all works well. VS Code finds `WebCmdletStrings.ProxyUriNotSupplied` and jump by F12:
![image](https://user-images.githubusercontent.com/22290914/81895008-9beaa500-95ca-11ea-88dc-7534ad17a4de.png)

![image](https://user-images.githubusercontent.com/22290914/81894970-870e1180-95ca-11ea-8c1b-1c4c6cdd9db2.png)

Reply:
It seems I can repo. It seems it is OmniSharp issue. After I restart VS Code I see the reds. Then I run Start-PSBuild and the reds goes away.

Reply:
I did run `Start-PSBuild -Clean` before starting VS Code. Also, we need to make sure opening VS Code with each .csproj folder should work too, for example, opening VS Code with the `src/System.Management.Automation` folder.

Reply:
I see the same. Open SMA folder - I see the reds. Run dotnet build from the folder - the reds goes away.
It is OmniSharp. I can open new issue there.

Reply:
@iSazonov are you running from `Start-PSBuild -Clean`? I'm not seeing OmniSharp pick up any of the generated classes after a clean build. 😕

Reply:
After switching to the branch I run `Start-PSBuild -Clean`, then run VS Code. After that I run `Start-PSBuild` and select a file.